### PR TITLE
Remove unused workflow files

### PR DESCRIPTION
**Pull request by iwahbe**
This PR removes `${PROVIDER}/config.yaml` files, since they are no longer used as the source of truth for the values they contain.

Previously, the set of providers was calculated by looking at the set of directories, which obviously will no longer work. Instead we just store the (previously calculated) JSON list directly.

### CODE REVIEW

File: provider-ci/Makefile [hunk 1]
```diff
@@ -30,6 +30,11 @@ test:
 format:
 	go fmt ./...
 
+providers/%/config.yaml: F=providers/$*/config.yaml
+providers/%/config.yaml:
+	mkdir -p $(dir $F)
+	curl "https://raw.githubusercontent.com/pulumi/pulumi-$*/master/.ci-mgmt.yaml" > $F
```
Comment (t0yv0) [inline provider-ci/Makefile:36]: What if the branch is `main`?

**Review by t0yv0 — APPROVED** [review]
LGTM with one nit.

File: provider-ci/Makefile (hunk 1 above)
Comment (iwahbe) [reply provider-ci/Makefile:36]: I just tested that. Github re-interprets `master` as `main`. I can make it explicit.

Comment (t0yv0) [reply provider-ci/Makefile]: Oh nice of them!

Comment (t0yv0) [reply provider-ci/Makefile]: no need then.

**Pull request merged**